src/LayerEditor/ui/mainwindow.py

```diff
@@ -268,9 +268,11 @@ def set_topology(self):
         self.regions.set_topology(cfg.diagram.topology_idx)
         
     def _update_regions(self):
-        """Update region for set shape"""
-        regions =self.diagramScene.selection.get_selected_regions(cfg.diagram)
-        self.regions.select_current_regions(regions)
+        """Update region panel, eventually set tab according to the selection in diagram"""
+        regions = self.diagramScene.selection.get_selected_regions(cfg.diagram)
+        if regions[0]:
```

Comment:
- 'select_current_regions' should be named 'select_current_region' and just single region should be passed
- For selection of single region, this should be selected in region panel. For more selected regions it was proposed to display None region, but regio of the last selected-deselected object can be used.

Reply:
That is an original function out of this pull request scope. However, a new branch is created for that pull request purpose.